src/s3ql/metadata.py

```diff
@@ -133,7 +133,7 @@ def cycle_metadata(backend, keep=10):
         pass
     cycle_fn("s3ql_metadata_new", "s3ql_metadata")
 
-    if cycle_fn is backend.copy:
+    if not backend.has_native_rename:
         backend.delete('s3ql_metadata_new')
```

Comment:
If I remember correctly, i found that the test `cycle_fn is backend.copy` returned False even when it was actually set to `backend.copy` in lines 115-118, which is why i changed it to checking `backend.has_native_rename`, which produced the correct result.

This change did make a difference in my setup. i used a local S3QL in an acd_cli mount point. Before this change, i would have a `s3ql_metadata_new` file left over after running `mkfs.s3ql` and after the change this temporary file would be correctly deleted.